W-19526870-agent-fabric-duke (#412)
mulesoft/docs-monitoring: modules/ROOT/pages/anypoint-insights-agentic.adoc

* MCP server: A service that implements the Model Context Protocol to expose tools and data to AI clients, enabling LLMs to invoke external capabilities through a standard interface.

This example shows metrics for an instance of an agent. 

[[dashboard-overview]]
//don't change order of numbers in graphic or calloutlist
image::metrics-agent.png["Metrics for an instance of an agent in Anypoint Monitoring"]

[calloutlist]
. Name of the instance
. <<properties, Properties of the instance>> that match properties in API Manager
. <<agent, Agent metrics>>
. <<lookback, Lookback period>> selected for the agent
. <<refresh, Auto-refresh>> (subscription tier requirement)
. Manual refresh option

For information about Agent Visualizer, see xref:agent-visualizer::index.adoc[]. 

== Find Metrics for Agentic Assets

Agent Visualizer links to metrics for agentic assets (agent, broker, and MCP server instances) from nodes in the visualizer graph. 

. In Agent Visualizer, click a node to open its detail card.
. In *Performance and Health*, click *View metrics* to view additional metrics in Anypoint Monitoring. 

> .. Set the lookback period to different range.

Author: ```suggestion
.. Optionally, set the lookback period to different range.
```

For more information about nodes in Agent Visualizer, see xref:agent-visualizer::index.adoc[].

//TODO: PREREQS: sub tier, permissions, instances on API Manager or Agent Visualizer? anything else?

[[lookback]]
== Set Lookback Period for Metrics

After navigating to the metrics for an agent, broker, or MCP server instance from Agent Visualizer, get up to the last 30 days of metrics on the instance. 

* Default: *Past 1 hour*
* Maximum: *Past 30 days*

//Available periods are identical to the periods available for Anypoint Insights. 

[[properties]]
== Instance Properties

//TODO: WHERE DO THE LINKS to Runtime Target, Implementation URI, and Proxy URL go?

Agent, broker, and MCP server pages display these properties:

include::partial$include-insights-agentic-details.adoc[tag=metadata]

[[agent]]
== Agent Metrics 

Agents protected by a Flex Gateway ingress policy (xref:policies::policies-availability-by-gateway.adoc[A2A Agent Card]) produce metrics in these charts:

* *Total Received*: All requests to the agent
* *Failed Requests*: Number of requests to the agent that failed
* *Failed Request Rate*: Percentage of requests to the agent that failed 

> 

Author: ```suggestion

The agent's Flex Gateway ingress supports request counts. The charts reflect calls to the agent through that ingress.
```

= Monitoring Agentic Assets

//TODO: CHECK IF API MANAGER OR ANY OTHER PRODUCT OPENS TO THIS PAGE
In US and EU Cloud instances of Anypoint Platform, navigate from Agent Visualizer to view detailed metrics on agentic assets (agent, broker, or MCP server instances).

> * Agent: An autonomous software component that uses goals, context, and available tools (often via an LLM) to decide and execute actions on behalf of a user or system.

Reviewer: You could use a DL here. It's the perfect use case.

Author: future

* MCP server: A service that implements the Model Context Protocol to expose tools and data to AI clients, enabling LLMs to invoke external capabilities through a standard interface.

This example shows metrics for an instance of an agent. 

[[dashboard-overview]]
//don't change order of numbers in graphic or calloutlist
image::metrics-agent.png["Metrics for an instance of an agent in Anypoint Monitoring"]

[calloutlist]
. Name of the instance
. <<properties, Properties of the instance>> that match properties in API Manager
. <<agent, Agent metrics>>
. <<lookback, Lookback period>> selected for the agent
. <<refresh, Auto-refresh>> (subscription tier requirement)
. Manual refresh option

For information about Agent Visualizer, see xref:agent-visualizer::index.adoc[]. 

== Find Metrics for Agentic Assets

Agent Visualizer links to metrics for agentic assets (agent, broker, and MCP server instances) from nodes in the visualizer graph. 

. In Agent Visualizer, click a node to open its detail card.
. In *Performance and Health*, click *View metrics* to view additional metrics in Anypoint Monitoring. 

> . Optional:

Reviewer: Suggest making this a bullet list as part of the previous step.

> . Optional:

Author: ```suggestion
```

//TODO: PREREQS: sub tier, permissions, instances on API Manager or Agent Visualizer? anything else?

[[lookback]]
== Set Lookback Period for Metrics

After navigating to the metrics for an agent, broker, or MCP server instance from Agent Visualizer, get up to the last 30 days of metrics on the instance. 

* Default: *Past 1 hour*
* Maximum: *Past 30 days*

//Available periods are identical to the periods available for Anypoint Insights. 

[[properties]]
== Instance Properties

//TODO: WHERE DO THE LINKS to Runtime Target, Implementation URI, and Proxy URL go?

Agent, broker, and MCP server pages display these properties:

include::partial$include-insights-agentic-details.adoc[tag=metadata]

[[agent]]
== Agent Metrics 

Agents protected by a Flex Gateway ingress policy (xref:policies::policies-availability-by-gateway.adoc[A2A Agent Card]) produce metrics in these charts:

* *Total Received*: All requests to the agent
* *Failed Requests*: Number of requests to the agent that failed
* *Failed Request Rate*: Percentage of requests to the agent that failed 

The agent's Flex Gateway ingress supports request counts. The charts reflect calls to the agent through that ingress.

> [[broker]]

Reviewer: ```suggestion

[[broker]]
```

== Broker Metrics

Metrics reported for brokers appear in these charts:

* *Requests Received*:
** *Total Received*: Number of requests to the broker
** *Failed Requests*: Number of requests to the broker that failed
** *Failed Request Rate*: Percentage of requests to the broker that failed 

////
NOT FOR GA
* *Calls Sent*:
* *A2A Calls Sent*: Number of requests the broker sent to agents
* *MCP Calls Sent*: Number of requests the broker sent to MCP servers
////

[[mcp-server]]
== MCP Server Metrics

MCP servers protected by a Flex Gateway ingress policy (xref:policies::policies-availability-by-gateway.adoc[MCP Support]) produce metrics in these charts:

> * *Total Received*: All requests to the agent
> * *Failed Requests*: Number of requests to the agent that failed
> * *Failed Request Rate*: Percentage of requests to the agent that failed 

Author: ```suggestion
* *Total Received*: All requests to the MCP server
* *Failed Requests*: Number of requests to the MCP server that failed
* *Failed Request Rate*: Percentage of requests to the MCP server that failed 
```

The MCP server's Flex Gateway ingress supports request counts. The charts reflect calls to the MCP server through that ingress.

[[refresh]]

> == Refreshing Metrics

Reviewer: ```suggestion
== Refresh Metrics
```

= Monitoring Agentic Assets

//TODO: VERIFY API MANAGER

> In US and EU Cloud instances of Anypoint Platform, navigate from Agent Visualizer or API Manager to view detailed metrics on agentic assets (agent, broker, or MCP server instances).

Author: @sathishpaulleo Can you tell me if the links to the details pages are from Visualizer and API Manager or other things? 

```suggestion
//TODO: VERIFY API MANAGER
In US and EU Cloud instances of Anypoint Platform, navigate from Agent Visualizer to view detailed metrics on agentic assets (agent, broker, or MCP server instances).
```

> In US and EU Cloud instances of Anypoint Platform, navigate from Agent Visualizer or API Manager to view detailed metrics on agentic assets (agent, broker, or MCP server instances).

Author: per Rajesh only Visualizer

```suggestion
In US and EU Cloud instances of Anypoint Platform, navigate from Agent Visualizer to view detailed metrics on agentic assets (agent, broker, or MCP server instances).
```

* Agent: An autonomous software component that uses goals, context, and available tools (often via an LLM) to decide and execute actions on behalf of a user or system.
* Broker: An intelligent routing service that coordinates task delegation across specialized agents in your enterprise. Brokers originate from MuleSoft and are referenced in the agent network YAML. 
* MCP server: A service that implements the Model Context Protocol to expose tools and data to AI clients, enabling LLMs to invoke external capabilities through a standard interface.

For information about Agent Visualizer, see xref:agent-visualizer::index.adoc[]. 

> //TODO: PREREQS: sub tier, permissions, instances on API Manager or Agent Visualizer? anything else?

Author: @sathishpaulleo For instance details, can you let me know if there are any special subscriptions or permissions to view this data?

```suggestion
//TODO: PREREQS: sub tier, permissions, instances on API Manager or Agent Visualizer? anything else?
```

Author: all tiers, Monitoring Viewer, ask Bhavin which API Manager Viewer permission

> //TODO: LOOKBACK settings 24 hours, anything else?

Author: @sathishpaulleo For the lookback, can you let me know what periods we allow? All that's in the mock is 24 hours.

```suggestion
//TODO: LOOKBACK settings 24 hours, anything else?
```

Author: same as Insights

> //TODO: AUTO-REFRESH INTERVALS 

Author: @sathishpaulleo For the auto-refresh interval options, can you let me know what we allow? Mock doesn't say.

```suggestion
//TODO: AUTO-REFRESH INTERVALS 
```

Author: Same as Insights (requires Titanium/Advanced) 

Only manual refresh if don't have Titanium/Advanced

//TODO: MANUAL REFRESH

//TODO: PROVIDE LINKS TO DOCS WITH INFO ON HOW TO OPEN THE METRICS PAGES

[[properties]]
== Instance Properties

> //TODO: WHERE DO THE LINKS to Runtime Target, Implementation URI, and Proxy URL go?

Author: @sathishpaulleo  For instance properties (metadata on the pages), can you help with where the links go. Mock doesn't provide info.

```suggestion
//TODO: WHERE DO THE LINKS to Runtime Target, Implementation URI, and Proxy URL go?
```

Author: Links are not to products but to apps, apis customers deploy. Technically links, but might not work.
